src/Common/src/System/Net/SocketAddress.cs

```diff
@@ -141,7 +139,6 @@ internal IPAddress GetIPAddress()
             }
             else if (Family == AddressFamily.InterNetwork)
             {
-                Debug.Assert(Size >= IPv4AddressSize);
```

Comment:
Were these asserts tripping?  They seem reasonable (well, I'd expect strict equality...)

Reply:
The following tests could trigger this assertion:
```cs


public static IEnumerable<object[]> Create_DefaultAddress_TestData()
{
    yield return new object[] { IPAddress.Parse("192.169.0.9"), 8, IPAddress.Any };
    yield return new object[] { IPAddress.Parse("192.169.0.9"), 15, IPAddress.Any };
    yield return new object[] { IPAddress.Parse("192.169.0.9"), 16, IPAddress.Any };
    yield return new object[] { IPAddress.Parse("192.169.0.9"), 32, IPAddress.Any };
    yield return new object[] { IPAddress.Parse("0:0:0:0:0:0:0:1"), 16, IPAddress.IPv6Any };
    yield return new object[] { IPAddress.Parse("0:0:0:0:0:0:0:1"), 28, IPAddress.IPv6Any };
    yield return new object[] { IPAddress.Parse("0:0:0:0:0:0:0:1"), 32, IPAddress.IPv6Any };
}

[Theory]
[MemberData(nameof(Create_DefaultAddress_TestData))]
public void Create_DefaultAddress_Success(IPAddress address, int size, IPAddress expectedAddress)
{
    var endPoint = new IPEndPoint(address, 500);
    var socketAddress = new SocketAddress(address.AddressFamily, size);

    IPEndPoint createdEndPoint = Assert.IsType<IPEndPoint>(endPoint.Create(socketAddress));
    Assert.NotSame(endPoint, createdEndPoint);
    Assert.Equal(expectedAddress, createdEndPoint.Address);
    Assert.Equal(expectedAddress.AddressFamily, createdEndPoint.AddressFamily);
    Assert.Equal(0, createdEndPoint.Port);
}

public static IEnumerable<object[]> Create_SmallIPV6Address_TestData()
{
    yield return new object[] { 8 };
    yield return new object[] { 15 };
}

[Theory]
[MemberData(nameof(Create_SmallIPV6Address_TestData))]
public void Create_SmallIPV6Address_ThrowsIndexOutOfRangeException(int size)
{
    IPAddress address = IPAddress.Parse("0:0:0:0:0:0:0:1");
    var endPoint = new IPEndPoint(address, 500);
    var socketAddress = new SocketAddress(address.AddressFamily, size);

    Assert.Throws<IndexOutOfRangeException>(() => Assert.IsType<IPEndPoint>(endPoint.Create(socketAddress)));
}
```

But: there appears to be a behaviour difference between Unix and Windows.

For example, Windows throws IndexOutOfRangeException for Create with an IPV6 address of size up to (and not including) 28. However, Unix throws IndexOutOfRangeExeption for Create with an IPV6 address of size up to (and not including) 16 - the size of an IPV4 address. 

This is visible in the test failures on this PR.

What do you think we should do? I'm leaning towards throwing an exception (AORE?) if Size < IPv4AddressSize or Size < IPv6AddressSize, but this could be breaking.

I think we should make Windows and Unix consistent here (they currently aren't). So therefore I'll wait on making them consistent till we decide what behaviour to standardize

Reply:
If we're already throwing exceptions but they're not consistent across platforms, then that's a bug that should be fixed and is necessarily breaking.  I think I'd personally just go with ArgumentException, but someone from networking should weigh in first.

Reply:
@bartonjs any progress with the exception? :)

Reply:
>If we're already throwing exceptions but they're not consistent across platforms, then that's a bug that should be fixed and is necessarily breaking. I think I'd personally just go with ArgumentException, but someone from networking should weigh in first.

If we're already throwing ArgumentOutOfRangeException we should continue to do that. It sounds like the difference in platforms (between Windows and Unix) is the calculation of the range value that trips the exception. But I don't understand what the current code is doing now. Is it allowing more invalid values on Windows or Unix?

Reply:
Previously for values:
- `size` <= 2 (IpV4 and IpV6): throws `ArgumentOutOfRangeException ` in `new SocketAddress(size)`
- 2 < `size` < 8 (IpV4 and IpV6): throws `ArgumentException` in `IpEndPoint.Create`
- 8 <= `size` < 16 (IpV4 and IpV6): hits assert in debug, throws `IndexOutOfRangeException` in release
- 16 <= `size` < 28 (IpV4): success
- 16 <= `size` < 28 (IpV6): hits assert in debug, throws `IndexOutOfRangeException` in release on Windows, succeeds on Mac
- 28 < `size` < `int.MaxValue` (IpV4 and IpV6): success

With this change:
- `size` <= 2 (IpV4 and IpV6): throws `ArgumentOutOfRangeException ` in `new SocketAddress(size)`
- 2 < `size` < 8 (IpV4 and IpV6): throws `ArgumentException` in `IpEndPoint.Create`
- 8 <= `size` < 16 (IpV4 and IpV6): throws `ArgumentException` in `IpEndPoint.Create`
- 16 <= `size` < 28 (IpV4): success
- 16 <= `size` < 28 (IpV6): throws `ArgumentException` in `IpEndPoint.Create`
- 28 < `size` < `int.MaxValue` (IpV4 and IpV6): success